### PR TITLE
Added correspondence_type migration with seeds.

### PR DESCRIPTION
**Pull request by Rnmarshall93**
Resolves [Create Correspondence Type Lookup Database Table](https://jira.devops.va.gov/browse/APPEALS-29187)

# Description
Added a migration for correspondence_table and added seed data as well.

## Acceptance Criteria
- [X] Code compiles correctly

## Testing Plan

1. Open a rails console
2. Execute the following commands
3. Run a migration to update the table changes.
```
require Rails.root.join("db/seeds/base.rb").to_s

Dir[Rails.root.join("db/seeds/*.rb")].sort.each { |f| require f }

Seeds::CorrespondenceTypes.new.seed!
```

4. You should now have the correct Correspondence_Type data in the table.
5. You can verify the correspondence_type data with `CorrespondenceType.all`

# Backend
## Database Changes
*Only for Schema Changes*

* [X] Add typical timestamps (`created_at`, `updated_at`) for new tables
* [ X] Update column comments; include a "PII" prefix to indicate definite or potential [PII data content](https://github.com/department-of-veterans-affairs/appeals-team/blob/master/caseflow-team/0-how-we-work/pii-handbook.md#what-is-pii)
* [X] Have your migration classes inherit from `Caseflow::Migration`, especially when adding indexes (use `add_safe_index`) (see [Writing DB migrations](https://github.com/department-of-veterans-affairs/caseflow/wiki/Writing-DB-migrations))
* [X] Verify that `migrate:rollback` works as desired ([`change` supported functions](https://edgeguides.rubyonrails.org/active_record_migrations.html#using-the-change-method))
* [ ] Perform query profiling (eyeball Rails log, check bullet and fasterer output)
* [ ] For queries using raw sql was an explain plan run by System Team
* [ ] Add appropriate indexes (especially for foreign keys, polymorphic columns, unique constraints, and Rails scopes)
* [ ] Run `make check-fks`; add any missing foreign keys or add to `config/initializers/immigrant.rb` (see [Record associations and Foreign Keys](https://github.com/department-of-veterans-affairs/caseflow/wiki/Record-associations-and-Foreign-Keys))
* [X] Add `belongs_to` for associations to enable the [schema diagrams](https://department-of-veterans-affairs.github.io/caseflow/task_trees/schema/schema_diagrams) to be automatically updated
* [ ] Document any non-obvious semantics or logic useful for interpreting database data at [Caseflow Data Model and Dictionary](https://github.com/department-of-veterans-affairs/caseflow/wiki/Caseflow-Data-Model-and-Dictionary)

### Test Coverage
Did you include any test coverage for your code? Check below:
- [ ] RSpec
- [ ] Jest
- [ ] Other

### CODE REVIEW

File: db/seeds/correspondence_types.rb [hunk 1]
```diff
@@ -0,0 +1,27 @@
...
+      correspondence_types_list =
+        ["Abeyance", "Attorney Inquiry", "CAVC Correspondence", "Change of address",
+         "Congressional interest", "CUE related", " Death certificate", "Evidence or argument", "Extension request",
+         " FOIA request", "Hearing Postponement Request", "Hearing related", "Hearing Withdrawal Request",
+         "Advance on docket", "Motion for reconsideration", "Motion to vacate", "Other motions",
+         "Power of attorney related", "Privacy Act complaints", "Privacy Act request", "Returned as undeliverable mail",
+         "Status Inquiry", "Thurber", "Withdrawal of appeal"]
+
+      correspondence_types_list.each do |type|
+        CorrespondenceType.find_or_create_by(name: type)
```
Comment (tradesmanhelix) [inline db/seeds/correspondence_types.rb:23]: 💯

File: app/models/correspondence_type.rb [hunk 2]
```diff
@@ -0,0 +1,3 @@
+class CorrespondenceType < ApplicationRecord
+  has_many :correspondence
```
Comment (tradesmanhelix) [inline app/models/correspondence_type.rb:2]: Should probably be the plural form: https://guides.rubyonrails.org/association_basics.html#the-has-many-association

File: db/seeds/correspondence_types.rb [hunk 3]
```diff
@@ -0,0 +1,27 @@
...
+    private
+
+    def create_correspondence_types
+      correspondence_types_list =
+        ["Abeyance", "Attorney Inquiry", "CAVC Correspondence", "Change of address",
+         "Congressional interest", "CUE related", " Death certificate", "Evidence or argument", "Extension request",
+         " FOIA request", "Hearing Postponement Request", "Hearing related", "Hearing Withdrawal Request",
+         "Advance on docket", "Motion for reconsideration", "Motion to vacate", "Other motions",
+         "Power of attorney related", "Privacy Act complaints", "Privacy Act request", "Returned as undeliverable mail",
+         "Status Inquiry", "Thurber", "Withdrawal of appeal"]
```
Comment (tradesmanhelix) [inline db/seeds/correspondence_types.rb:20]: Looks like there might be an extra space in `" Death certificate"`.

Also, might not be a bad idea to put these in their own method, one per line, just to help with readability 🤷 

```
def create_correspondence_types
  correspondence_types_list = correspondence_types_list
  ...
end

def correspondence_types_list
  [
    # items here
  ]
end
```

**Review by tradesmanhelix — COMMENTED** [review]
A couple minor things but overall looking great!! 👍

**Review by jimfoley25 — DISMISSED** [review]
Looks good

**Review by HunJerBAH — APPROVED** [review]

**Review by HunJerBAH — APPROVED** [review]

**Pull request merged**